src/omni_epd/displays/inky_display.py

```diff
@@ -33,123 +34,77 @@ class InkyDisplay(VirtualEPD):
-                      "what_black", "what_red", "what_yellow"]
-
-        # python libs for this might not be installed - that's ok, return nothing
-        if(check_module_installed(INKY_PKG)):
-            # if passed return list of devices
-            result = [f"{INKY_PKG}.{n}" for n in deviceList]
-
-        return result
-
-    def _display(self, image):
-        # apply any needed conversions to this image based on the mode
-        image = self._filterImage(image)
-
-        # set the image and display
-        self._device.set_border(getattr(self._device, self._get_device_option('border', '').upper(), self._device.border_colour))
-        self._device.set_image(image)
-        self._device.show()
-
-    def clear(self):
-        for _ in range(2):
-            for y in range(self.height - 1):
-                for x in range(self.width - 1):
-                    self._device.set_pixel(x, y, self._device.WHITE)
-
-        self._device.show()
-
-
-class InkyImpressionDisplay(VirtualEPD):
-    """
-    This is an abstraction for Pimoroni Inky Impression devices
-    https://shop.pimoroni.com/products/inky-impression
-    https://github.com/pimoroni/inky
-    """
-
-    pkg_name = INKY_PKG
-    mode = 'color'  # this uses color by default
-    max_colors = 8  # 7 + CLEAN (no color)
-    modes_available = ('bw', 'color')
-
-    def __init__(self, deviceName, config):
-        super().__init__(deviceName, config)
-
-        # load the device driver
-        deviceObj = self.load_display_driver(self.pkg_name, 'inky_uc8159')
-        self._device = deviceObj.Inky()
-
-        # set the width and height
-        self.width = self._device.width
-        self.height = self._device.height
-
-        # get colors from the inky lib (won't be used normally as inky does conversion)
-        self.palette_filter = deviceObj.DESATURATED_PALETTE
+    def load_device(self, deviceName):
+        # need to figure out what type of device we have
+        dType, dColor, *_ = deviceName.split('_') + [None]
+        if(dType == 'phat'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'phat')
+            device = deviceObj.InkyPHAT(dColor)
+        elif(dType == 'phat1608'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'phat')
+            device = deviceObj.InkyPHAT_SSD1608(dColor)
+        elif(dType == 'what'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'what')
+            device = deviceObj.InkyWHAT(dColor)
+        elif(dType == 'impression'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'inky_uc8159')
+            device = deviceObj.Inky()
+        elif(dType == 'auto'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'auto')
+            device = deviceObj.auto()
+
+        if(device.colour == 'multi'):
+            return device, device.CLEAN, 'color'
+        else:
+            return device, device.WHITE, dColor

```

Comment:
```suggestion
            return device, device.WHITE, device.colour

```

Reply:
We'd still need the dColor variable when the user is specifying the exact device (ie, `inky.phat_red`) but once the driver is loaded either manually or via auto the device should always know it's own color and return the right value. This will prevent the color mode from being `None` on non-Impression devices.